src/main/python/resources/annotaters/vep/domain.py

```diff
@@ -0,0 +1,88 @@
+import pandas as pd
+from src.main.python.resources.annotaters.vep.template import Template
+import numpy as np
+
+
+class Domain(Template):
+    """
+    This class contains a lot of complex Pandas codebase,
+    allow me to explain here:
+
+    First the column Domain.name is split on the '&' character,
+    each split gets it's own column due to expand=True.
+    Then another split happens on ":" to split the source from the name,
+    where only the source is kept.
+    Then the function process_ndomain is called to set the index value 3 to any
+    column that contains the characters: '_domain' or '_profile',
+    given that they are not NaN.
+    If a column does not contain the characters, the value is left unchanged.
+    Then I know that the remaining terms should directly match every key in the
+    property levels_dict, so a direct replace is called for that.
+    The tricky part is then to mark everything that is left and not NaN as 5,
+    which is what the function process_others does.
+    It looks if a value is
+        1. not NaN,
+        2. not within the levels_dict keys and
+        3. not within the levels_dict values.
+    Anything that is left "True" after these 3 filters is set to 5.
+    Finally the index numbers are aggregated over the columns with the 'min'
+    function, mapped back to the string and applied.
+    """
+
+    def __init__(self):
+        super(Domain, self).__init__(
+            name='DOMAINS',
+            usable=True
+        )
+
+    @property
+    def columns(self):
+        return ['Domain']
+
+    @property
+    def levels_dict(self):
+        return {'PANTHER': 4, 'ndomain': 3, 'Low_complexity_(Seg)': 2,
+                'Cleavage_site_(Signalp)': 1, 'Coiled-coils_(Ncoils)': 0}
+
+    @property
+    def output_dict(self):
+        return {0.0: 'ncoils', 1.0: 'sigp', 2.0: 'lcompl', 3.0: 'ndomain',
+                4.0: 'hmmpanther', 5.0: 'other'}
```

Comment:
Unknown. Keys of `output_dict(self)` could just be `int` instead of `float` ;)

Reply:
In that case it might be just a tad cleaner to make these identical (initially wanted to suggest an Enum which could return the 2 different Strings, but not sure how complicated that would be within Python).

Reply:
Enums are available for python, although I'm not very experienced with them, it's something that can be tried.

Reply:
Not very known with the Python enums. In Java I couldn't live without (custom method implementation based on enum type is awesome). I did came across [this](https://stackoverflow.com/a/35567824). Perhaps it's worth trying out? (though it's probably rather low-priority issue)